corehq/doctypemigrations/bulk_migrate.py

```diff
@@ -5,37 +5,25 @@
 from corehq.util.couch import IterDB
 
 
-def fetch_binary_attachment(db, doc_id, name):
-    data = db.fetch_attachment(doc_id, name)
+def _encode_data(attachment_dict):
+    data = attachment_dict['data']
     # couchdbkit is annoying and auto-converts
     # anything that looks like utf-8 to unicode
     if isinstance(data, unicode):
         data = data.encode('utf-8')
-    return data
+    attachment_dict['data'] = base64.b64encode(data)
+    return attachment_dict
 
 
-def insert_attachment(db, doc_json):
-    """
-    fetch attachment bodies for doc_json with attachment stubs
-    and insert them into the doc_json
-
-    """
+def _insert_attachments(db, doc_json):
     if '_attachments' not in doc_json:
         return
     else:
-        for name, value in doc_json['_attachments'].items():
-            # it's a bit of a wash whether to fetch attachments individually
-            # or re-fetch the whole doc with attachments.
-            # For a doc with a single attachment this is faster,
-            # so I'll go with it for now
-            data = fetch_binary_attachment(db, doc_json['_id'], name)
-            doc_json['_attachments'][name] = {
-                'data': base64.b64encode(data),
-                'length': value['length'],
-                'content_type': value['content_type'],
-                'digest': value['digest'],
-                'revpos': value['revpos']
-            }
+        full_doc = db.get(doc_json['_id'], attachments=True)
+        doc_json['_attachments'] = {
+            name: _encode_data(attachment_dict)
+            for name, attachment_dict in full_doc['_attachments'].items()
+        }
```

Comment:
Ah ok, my guess is then that this is double encoding. And now that I think of it, the other code was docs directly from the view with attachments and saving the JSON as-is, which further points in that direction.

The good news is that double encoding was probably making your time slower that it should be (since base64 inflates size of a payload by 33%).